tests/functional/aws-node-sdk/test/multipleBackend/put.js

```diff
@@ -113,6 +151,118 @@ describe('MultipleBackend put object', () => {
                     });
                 });
             });
+
+            it('should put an object to AWS', done => {
+                const params = { Bucket: bucket, Key: key,
+                    Body: body,
+                    Metadata: { 'scal-location-constraint': 'aws-test' } };
+                s3.putObject(params, err => {
+                    assert.equal(err, null, 'Expected success, ' +
+                        `got error ${err}`);
+                    s3.getObject({ Bucket: bucket, Key: key }, (err, res) => {
+                        assert.strictEqual(err, null, 'Expected success, ' +
+                        `got error ${err}`);
+                        assert.strictEqual(res.ETag, `"${correctMD5}"`);
+                        done();
+                    });
+                });
+            });
+
+            itSkipIfE2E('should put objects with same key to AWS then file, ' +
+            'and object should only be present in file', done => {
+                const params = { Bucket: bucket, Key: key,
+                    Body: body,
+                    Metadata: { 'scal-location-constraint': 'aws-test' } };
+                s3.putObject(params, err => {
+                    assert.equal(err, null, 'Expected success, ' +
+                        `got error ${err}`);
+                    params.Metadata = { 'scal-location-constraint': 'file' };
+                    s3.putObject(params, err => {
+                        assert.equal(err, null, 'Expected success, ' +
+                            `got error ${err}`);
+                        s3.getObject({ Bucket: bucket, Key: key },
+                        (err, res) => {
+                            assert.equal(err, null, 'Expected success, ' +
+                                `got error ${err}`);
+                            assert.strictEqual(
+                                res.Metadata['scal-location-constraint'],
+                                'file');
+                            awsS3.getObject({ Bucket: 'multitester555',
+                            Key: key }, err => {
+                                assert.strictEqual(err.code, 'NoSuchKey');
+                                done();
+                            });
+                        });
+                    });
+                });
+            });
+
+            itSkipIfE2E('should put objects with same key to file then AWS, ' +
+            'and object should only be present on AWS', done => {
+                const params = { Bucket: bucket, Key: key,
+                    Body: body,
+                    Metadata: { 'scal-location-constraint': 'file' } };
+                s3.putObject(params, err => {
```

Comment:
You may use `async.series` to chain the calls in a more readable way and decrease indentation level. (same for other tests of similar shape).